myjdapi/myjdapi.py

```diff
@@ -786,74 +819,81 @@ def get_device(self, device_name=None, device_id=None):
             for device in self.__devices:
                 if device["name"] == device_name:
                     return Jddevice(self, device)
-        raise(MYJDException("Device not found\n"))
+        raise (MYJDException("Device not found\n"))
 
-    def request_api(self, path, http_method="GET",params=None, action=None):
+    def request_api(self, path, http_method="GET", params=None, action=None, api=None):
         """
         Makes a request to the API to the 'path' using the 'http_method' with parameters,'params'.
         Ex:
         http_method=GET
         params={"test":"test"}
         post_params={"test2":"test2"}
         action=True
-        This would make a request to "http://api.jdownloader.org/"
+        This would make a request to "https://api.jdownloader.org"
         """
+        if not api:
+            api = self.__api_url
```

Comment:
This could be on the header of the function. Instead of:
` def request_api(self, path, http_method="GET", params=None, action=None, api=None):`
Use :
` def request_api(self, path, http_method="GET", params=None, action=None, api=self.__api_url):`
and you don't need this part you added.

Reply:
tried it but didn't work that's why crappy way mm..

    def request_api(self, path, http_method="GET", params=None, action=None, api=self.__api_url):
NameError: name 'self' is not defined

Reply:
Oh yes that's true. Ok no problem then.